extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/devmode/ArcEndpointRecorder.java

```diff
@@ -0,0 +1,131 @@
+package io.quarkus.vertx.http.runtime.devmode;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Type;
+import java.util.Iterator;
+import java.util.List;
+
+import javax.enterprise.inject.Any;
+import javax.enterprise.inject.Default;
+
+import io.quarkus.arc.InjectableBean;
+import io.quarkus.arc.InjectableObserverMethod;
+import io.quarkus.arc.impl.ArcContainerImpl;
+import io.quarkus.runtime.annotations.Recorder;
+import io.quarkus.vertx.http.runtime.devmode.Json.JsonArrayBuilder;
+import io.quarkus.vertx.http.runtime.devmode.Json.JsonObjectBuilder;
+import io.vertx.core.Handler;
+import io.vertx.ext.web.RoutingContext;
+
+@Recorder
+public class ArcEndpointRecorder {
+
+    public Handler<RoutingContext> createBeansHandler() {
+        return new Handler<RoutingContext>() {
+
+            @Override
+            public void handle(RoutingContext ctx) {
+                ctx.response().putHeader("Content-type", "application/json");
+
+                ArcContainerImpl container = ArcContainerImpl.instance();
+                List<InjectableBean<?>> beans = container.getBeans();
```

Comment:
Perhaps as a follow up to this, `ArcContainerImpl#getBeans` can return a `LinkedHashSet` so it can return data in the same order

Reply:
Hm, `getBeans()` returns a list...

Reply:
Ah damn it, I was looking in the IDE and was looking at `getBeans(String name)` :facepalm: